test/src/test/java/org/corfudb/infrastructure/log/StreamLogFilesTest.java

```diff
@@ -68,6 +71,145 @@ public void testWriteReadWithChecksum() {
+                entries.add(LogData.getTrimmed(x));
+            } else {
+                entries.add(getEntry(x));
+            }
+        }
+
+        log.append(entries);
+        long trimMarkAfterWrite = (long) (numEntries * trimmedRatio);
+        assertThat(log.getTrimMark()).isEqualTo(trimMarkAfterWrite);
+    }
+
+    @Test
+    public void testRangeWriteAfterPrefixTrim() throws Exception {
+        ServerContext sc = getContext();
+        StreamLog log = new StreamLogFiles(sc, false);
+        final long trimMark = 1000;
+        final long trimOverlap = 100;
+        final long numEntries = 200;
+        log.prefixTrim(trimMark);
+
+        List<LogData> entries = new ArrayList<>();
+        for (long x = 0; x < numEntries; x++) {
+            long address = trimMark - trimOverlap + x;
+            entries.add(getEntry(address));
+        }
+
+        log.append(entries);
+
+        StreamLog log2 = new StreamLogFiles(sc, false);
+        List<LogData> readEntries = readRange(0L, trimMark - trimOverlap + numEntries, log2);
+
+        List<LogData> notTrimmed = new ArrayList<>();
+        for (LogData entry : readEntries) {
+            if (!entry.isTrimmed()) {
+                notTrimmed.add(entry);
+            }
+        }
+
+        assertThat((long) notTrimmed.size()).isEqualTo(numEntries - trimOverlap - 1);
+    }
+
+    @Test
+    public void testRangeOverwrite() throws Exception {
+        ServerContext sc = getContext();
+        StreamLog log = new StreamLogFiles(sc, false);
+        final int numIter = 500;
+
+        List<LogData> entries = new ArrayList<>();
+        for (long x = 0; x < numIter; x++) {
+            entries.add(getEntry(x));
+        }
+
+        String logDir = sc.getServerConfig().get("--log-path") + File.separator + "log";
+        File file = new File(logDir);
+        // Write the same range twice
+        log.append(entries);
+        log.sync(true);
+        long logSize = FileUtils.sizeOfDirectory(file);
+        log.append(entries);
+        log.sync(true);
+        long logSize2 = FileUtils.sizeOfDirectory(file);
+        assertThat(logSize2).isEqualTo(logSize);
+    }
+
+    @Test
+    public void writingTrimmedEntries() throws Exception {
+        ServerContext sc = getContext();
+        StreamLog log = new StreamLogFiles(sc, false);
+
+        final long trimMark = 400;
+        final long numEntries = trimMark + 200;
+        log.prefixTrim(trimMark);
+
+        List<LogData> entries = new ArrayList<>();
+        for (int x = 0; x < numEntries; x++) {
+            entries.add(LogData.getTrimmed(x));
+        }
+
+        String logDir = sc.getServerConfig().get("--log-path") + File.separator + "log";
+        File file = new File(logDir);
+        long logSize = FileUtils.sizeOfDirectory(file);
+        log.append(entries);
+        log.sync(true);
+        long logSize2 = FileUtils.sizeOfDirectory(file);
+        assertThat(logSize2).isEqualTo(logSize);
+    }
```

Comment:
Do you need to test the exception conditions and their handling ? IllegalArgumentException is thrown in 2 places.